test/algebraic_simplification.cpp

```diff
@@ -596,3 +596,37 @@ TEST(algebraic_simplification, pass_property)
     ASSERT_TRUE(pass->get_property(pass::PassProperty::REQUIRE_STATIC_SHAPE));
     ASSERT_FALSE(pass->get_property(pass::PassProperty::CHANGE_DYNAMIC_STATE));
 }
+
+TEST(algebraic_simplification, replace_transpose_with_reshape_4d)
+{
+    Shape shape_in{1, 2, 3, 3};
+    auto param = make_shared<op::Parameter>(element::boolean, shape_in);
+    auto constant_perm =
+        make_shared<op::Constant>(element::i64, Shape{4}, vector<int64_t>{1, 0, 2, 3});
+    auto transpose = make_shared<op::Transpose>(param, constant_perm);
+    auto f = make_shared<Function>(transpose, ParameterVector{param});
+
+    pass::Manager pass_manager;
+    pass_manager.register_pass<pass::AlgebraicSimplification>();
+    pass_manager.run_passes(f);
+
+    ASSERT_EQ(count_ops_of_type<op::Transpose>(f), 0);
+    ASSERT_EQ(count_ops_of_type<op::Reshape>(f), 1);
+}
+
+TEST(algebraic_simplification, replace_transpose_with_reshape_3d)
```

Comment:
Need add much more unit tests:
- shape [10,20,1,1], permute_order = [0,2,3,1]
- shape [10,1,1,20], permute_order = [0,3,1,2]
- shape [10,20,1,2], permute_order = [0,2,1,3]
- shape [10,1,1,1,20], permute_order = [0,4,1,2,3]
- shape [10,20,1,1,1], permute_order = [0,2,3,4,1]
- shape [10,1,1,1,1], permute_order = [1,4,2,3,0]
- shape [10,1,1,1,1], permute_order = [4,2,0,1,3]
- shape [1,1,1,1,1], permute_order = [4,2,0,1,3]

Also add tests where the transformation should not be applied:
- shape [10,20,1,2], permute_order = [0,2,3,1]
- shape [10,20,1,2], permute_order = [0,3,1,2]
- shape [10,20], permute_order = [1,0]

Also need some tests where some dimensions are dynamic.

Reply:
Tested all the cases above manually, not sure if we need a test for every permuatation @diyessi ? , although I wouldn't mind,  and added a test for a failing case (in cases it shouldn't replace)

Reply:
Not sure what "manually" testing things is. I would add unit tests for all cases listed above. Unit tests for cases that pass makes sure they don't fail in the future when someone else edits the code.

Reply:
By manually, I meant I copied the shapes and made sure everything mentioned above passed. I added those cases now.